### PR TITLE
DQ Temporary solution.

### PR DESCRIPTION
**Pull request by Ksteeds**
Will aim to build out a more sophisticated solution now.

Used to understand how the simplest solution would perform.

### CODE REVIEW

**Comment by Ksteeds** [comment]
Will look to isolate the specific batch that failed to run, as opposed to calling the entire function again.

File: macrosynergy/management/dq.py [hunk 1]
```diff
@@ -336,19 +337,27 @@ def _request(self, endpoint: str, tickers: List[str], params: dict,
...
                     for elem in tick_list_compr[(i * 10):(i + 1) * 10]:
                         params["expressions"] = elem
                         results = executor.submit(self._fetch_threading, endpoint, params)
-                        time.sleep(0.75)
+                        time.sleep(delay)
                         output.append(results)
 
                     for f in concurrent.futures.as_completed(output):
                         try:
                             response = f.result()
```
Comment (lsimonsen) [inline macrosynergy/management/dq.py:352]:
```python
response = f.result()
time.sleep(delay)
```

File: macrosynergy/management/dq.py [hunk 2]
```diff
@@ -336,19 +337,27 @@ def _request(self, endpoint: str, tickers: List[str], params: dict,
         output = []
         if self.concurrent:
             for i in range(exterior_iterations):
-                if i > 0: time.sleep(0.3)
+                if i > 0:
+                    time.sleep(0.3)
```
Comment (lsimonsen) [inline macrosynergy/management/dq.py:342]: Remove?

File: macrosynergy/management/dq.py [hunk 3]
```diff
@@ -336,19 +337,27 @@ def _request(self, endpoint: str, tickers: List[str], params: dict,
...
                 with concurrent.futures.ThreadPoolExecutor() as executor:
                     for elem in tick_list_compr[(i * 10):(i + 1) * 10]:
                         params["expressions"] = elem
                         results = executor.submit(self._fetch_threading, endpoint, params)
-                        time.sleep(0.75)
+                        time.sleep(delay)
```
Comment (lsimonsen) [inline macrosynergy/management/dq.py:347]: Keep:
```python
time.sleep(delay)
```

File: macrosynergy/management/dq.py [hunk 4]
```diff
@@ -336,19 +337,27 @@ def _request(self, endpoint: str, tickers: List[str], params: dict,
         output = []
         if self.concurrent:
             for i in range(exterior_iterations):
```
Comment (lsimonsen) [inline macrosynergy/management/dq.py:340]: Remove exterior for loop

File: macrosynergy/management/dq.py [hunk 5]
```diff
@@ -336,19 +337,27 @@ def _request(self, endpoint: str, tickers: List[str], params: dict,
...
                             response = f.result()
                         except ValueError:
-                            raise ValueError("Server being hit too quickly with requests.")
+                            print("Server being hit too quickly with requests.")
+                            count += 1
+                            delay *= 1.5
+                            params.pop("expressions")
+                            final_output = self._request(endpoint="/expressions/time-series",
+                                                         tickers=tickers, params=params,
+                                                         delay=delay, count=count)
```
Comment (lsimonsen) [inline macrosynergy/management/dq.py]: Remove and re-instate `ValueError`

File: macrosynergy/management/dq.py [hunk 6]
```diff
@@ -336,19 +337,27 @@ def _request(self, endpoint: str, tickers: List[str], params: dict,
...
+                            return final_output
```
Comment (lsimonsen) [inline macrosynergy/management/dq.py:361]:
```python
tmp =  = self._request(endpoint="/expressions/time-series",
                                                         tickers=tickers, params=params,
                                                         delay=delay, count=count)
final_output.extend(tmp)
```

Remove `return final_output`

**Comment by lsimonsen** [comment]
@Ksteeds can we close this one as it has been superseeded by [PR 157](https://github.com/macrosynergy/macrosynergy/pull/157)?

**Review by lsimonsen — DISMISSED** [review]

**Review by lsimonsen — APPROVED** [review]

**Pull request merged**